demo/site/src/pages/preview/[[...path]].tsx

```diff
@@ -11,12 +31,51 @@ export default function AuthenticatedPreviewPage(props: InferGetServerSidePropsT
     );
 }
 
+export const createPagePath = (path: string | string[] | undefined) => {
+    if (!path) {
+        return "/";
+    }
+    if (Array.isArray(path)) {
+        return `/${path.join("/")}`;
+    }
+    return `/${path}`;
+};
+
 export const getServerSideProps: GetServerSideProps<PageUniversalProps> = async (context: GetServerSidePropsContext) => {
-    const { includeInvisibleBlocks } = parsePreviewState(context.query);
-    const getUniversalProps = createGetUniversalProps({
+    const { includeInvisibleBlocks } = parsePreviewParams(context.query);
+    const contentScope = { domain, language: defaultLanguage };
+    const path = createPagePath(context.params?.path);
+    const clientOptions = {
         includeInvisiblePages: true,
         includeInvisibleBlocks,
         previewDamUrls: true,
+    };
+    const graphqQLCLient = createGraphQLClient(clientOptions);
+
+    const data = await graphqQLCLient.request<GQLPageTypeQuery, GQLPageTypeQueryVariables>(pageTypeQuery, {
+        path,
+        contentScope,
     });
+
+    if (data.pageTreeNodeByPath?.documentType === "Link") {
+        const { pageTreeNode } = await graphqQLCLient.request<GQLPreviewLinkQuery, GQLPreviewLinkQueryVariables>(previewLinkQuery, {
+            id: data.pageTreeNodeByPath.id,
+        });
+
+        if (pageTreeNode?.document?.__typename === "Link") {
+            const destination = createLinkRedirectDestination(pageTreeNode.document.content);
+
+            if (destination) {
+                return {
+                    redirect: {
+                        destination,
+                        permanent: false,
```

Comment:
You'll probably need to set some cache headers here. Otherwise, the old redirect may still be used after changing the link target

Reply:
With permanent false the redirect won't be cached.

https://nextjs.org/docs/api-reference/next.config.js/redirects

"permanent true or false - if true will use the 308 status code which instructs clients/search engines to cache the redirect forever, **if false will use the 307 status code which is temporary and is not cached.**"